fs-cache/src/cache.rs

```diff
@@ -0,0 +1,668 @@
+use std::fs;
+use std::num::NonZeroUsize;
+use std::path::{Path, PathBuf};
+
+use data_error::{ArklibError, Result};
+use fs_atomic_light::temp_and_move;
+use fs_storage::utils::extract_key_from_file_path;
+use lru::LruCache;
+
+/// A cache entry that stores a value and its size in bytes.
+///
+/// This structure is used to track both the actual data (value)
+/// and its memory usage (size) in the cache.
+struct CacheEntry<V> {
+    value: V,
+    size: usize,
+}
+
+/// A combined in-memory and disk-based cache system.
+///
+/// This cache uses an LRU (Least Recently Used) eviction policy for the
+/// in-memory portion and persists data to disk for long-term storage.
+pub struct Cache<K, V> {
+    /// Label for logging
+    label: String,
+    /// Path to the underlying folder where data is persisted
+    path: PathBuf,
+    /// An in-memory LRU cache for quick access to frequently used items.
+    memory_cache: LruCache<K, CacheEntry<V>>,
+    /// The current memory usage in bytes.
+    current_memory_bytes: usize,
+    /// The maximum allowable memory usage in bytes.
+    max_memory_bytes: usize,
+    /// Whether to include values in debug logs
+    log_values: bool,
```

Comment:
The `log_values` field is only read once in the code (on line 95), and I see logging occurring without checking it in other parts. Is this particular log statement somehow different? 

I think we can safely assume it’s always enabled and remove this field entirely.